pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -537,6 +549,14 @@ def _is_slurm_managing_tasks(self) -> bool:
     def _choose_strategy(self) -> Union[Strategy, str]:
         if self._accelerator_flag == "ipu":
             return IPUStrategy.strategy_name
+        if self._accelerator_flag == "hpu":
+            if not _HPU_AVAILABLE:
+                raise MisconfigurationException("HPU Accelerator requires HPU devices to run")
+
```

Comment:
should we defer this exception to the HPU accelerator and/or strategy constructors? @four4fish ?

Reply:
This is not needed. HPUAccelerator.is_available() already called in `_set_parallel_devices_and_init_accelerator` L507

Reply:
Updated!